src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ErrorNotificationHandler.java

```diff
@@ -15,7 +16,7 @@
 import java.util.concurrent.Executor;
 import java.util.concurrent.Executors;
 
-public class ErrorNotificationHandler implements IMessageHandler {
+public class ErrorNotificationHandler implements ISessionHandler {
```

Comment:
This solution probably won't work, as `ISessionHandler` is for processing messages in sessions. We don't have sessions enabled in our queue and we don't need them.

I'd use IMessageHandler with `complete` and `deadLetter` functionalities passed to it, either as functions or in one interface. These methods/functions would point to the corresponding methods of the QueueClient in which the handler will be registered. Alternatively, IMessageReceiver could be used - it has the advantage of allowing us to pause reading messages, for example when some dependency is down.